portals/publisher/src/main/webapp/source/src/app/components/Apis/Details/Properties/Properties.jsx

```diff
@@ -418,6 +418,11 @@ function Properties(props) {
         if (additionalPropertiesCopy !== additionalProperties) {
             setIsAdditionalPropertiesStale(true);
         }
+        if (additionalPropertiesCopy.length === 0 && customPropertiesTemp?.length === 0) {
+            setIsAdditionalPropertiesStale(false);
+        } else if (additionalPropertiesCopy.length === 0 && customPropertiesTemp?.length !== 0) {
+            setShowAddProperty(true);
+        }
```

Comment:
Isn't it better to extract additionalPropertiesCopy.length === 0 check outside to avoid redundant checks?

Reply:
Fixed with https://github.com/wso2/apim-apps/pull/990/commits/a30e38872b821f81aa796da4572d8496c56bb729